ark/utils/test_utils.py

```diff
@@ -0,0 +1,168 @@
+    if sub_dir is None:
+        sub_dir = ""
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        filelocs[fov] = []
+        fov_path = os.path.join(base_dir, fov, sub_dir)
+        os.makedirs(fov_path)
+        for j, name in enumerate(img_names):
+            io.imsave(os.path.join(base_dir, name), tif_data[i, :, :, j])
+            filelocs[fov].append(os.path.join(base_dir, fov, sub_dir, name))
+
+    return filelocs, tif_data
+
+
+def _create_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, tif_data[i, :, :, :], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+def _create_mibitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    mass_map = tuple(enumerate(channel_names, 1))
+
+    for i, fov in enumerate(fov_names):
+        tif_obj = mi.MibiImage(tif_data[i, :, :, :],
+                               mass_map,
+                               **METADATA)
+
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        tiff.write(tiffpath, tif_obj, dtype=dtype)
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+TIFFMAKERS = {
+    'tiff': _create_tifs,
+    'multitiff': _create_multitiff,
+    'mibitiff': _create_mibitiff,
+}
+
+
+def create_paired_xarray_fovs(base_dir, fov_names, channel_names, img_shape=(1024, 1024),
+                              mode='tiff', delimiter=None, sub_dir=None, fills=None, dtype="int8"):
+
+    if not os.path.isdir(base_dir):
+        raise FileNotFoundError(f'{base_dir} is not a directory')
+        return None
+
+    if fov_names is None or fov_names is []:
+        raise ValueError('No fov names were given...')
+        return None
+
+    if channel_names is None or channel_names is []:
+        raise ValueError('No image names were given...')
+        return None
+
+    if not isinstance(fov_names, list):
+        fov_names = [fov_names]
+
+    if not isinstance(channel_names, list):
+        channel_names = [channel_names]
+
+    # verify fills structure -> dict(fov_name : list [num_channels])
+    if fills is not None:
+        if fills.keys() is not fov_names:
+            raise ValueError(f'fills keys, {fills.keys()}, must match fov_names, {fov_names}')
+            return None
+        for fill in fills.values():
+            if len(fill) != len(channel_names):
+                raise ValueError(f'Not all fills have length {len(channel_names)}')
+                return None
+
+    if fills is not None and not isinstance(fills, list):
+        fills = [fills]
+
+    if fills is not None and (len(fills) != len(channel_names)):
```

Comment:
Any reason you put parentheses around len(fills) != len(channel_names)?

Reply:
C/C++ habits die hard